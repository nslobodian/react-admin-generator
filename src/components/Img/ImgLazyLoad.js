import React from 'react'
import PropTypes from 'prop-types'
import LazyLoad from 'react-lazyload'
import Img from './Img'

const ImgLazyLoad = ({height = '100%', offset = 100, ...props}) =>
  <LazyLoad
    throttle={100}
    height={height}
    offset={offset}
    once
  >
    <Img {...props} />
  </LazyLoad>

ImgLazyLoad.propTypes = {
  props: PropTypes.object,
  height: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  offset: PropTypes.number,
}
export default ImgLazyLoad
