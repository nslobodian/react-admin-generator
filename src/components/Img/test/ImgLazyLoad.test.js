import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ImgLazyLoad from '../ImgLazyLoad'

describe('(Components) ImgLazyLoad', () => {
  it('Should render component properly - with default properties', () => {
    const props = {};
    const wrapper = shallow(
      <ImgLazyLoad
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  }),
  it('Should render component properly - with given properties', () => {
    const props = {};
    const height = 10;
    const offset = 2;
    const wrapper = shallow(
      <ImgLazyLoad
        {...props}
        height={height}
        offset={offset}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

