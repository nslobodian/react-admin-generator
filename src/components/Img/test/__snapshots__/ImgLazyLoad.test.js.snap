// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) ImgLazyLoad Should render component properly - with default properties 1`] = `
<LazyLoad
  height="100%"
  offset={100}
  once={true}
  overflow={false}
  resize={false}
  scroll={true}
  throttle={100}
  unmountIfInvisible={false}
>
  <_class />
</LazyLoad>
`;

exports[`(Components) ImgLazyLoad Should render component properly - with given properties 1`] = `
<LazyLoad
  height={10}
  offset={2}
  once={true}
  overflow={false}
  resize={false}
  scroll={true}
  throttle={100}
  unmountIfInvisible={false}
>
  <_class />
</LazyLoad>
`;
